### PR TITLE
Re-write cross-contract call documentation

### PR DESCRIPTION
**Pull request by HCastano**
This PR re-writes the cross-contract docs.

It improves the section on contract references by now walking readers through the newly
added `basic_contract_ref` example (see https://github.com/paritytech/ink/pull/1704).

It also adds a section on using `Builders` for cross-contract calls.

A future addition would be to mention the new `contract_ref!` macro for doing trait-based
cross-contract calls (see https://github.com/paritytech/ink/pull/1673).

### CODE REVIEW

File: docs/basics/cross-contract-calling.md [hunk 1]
```diff
@@ -6,99 +6,345 @@ hide_title: true
...
+
+A downside to using them is that you need to import the contract you want to call as a
+dependency of your own contract.
+
+If you want to interact with a contract that is already on-chain you will need to use the
+[`Builders`](/basics/cross-contract-calling#builders) approach instead.
+
+### `BasicContractRef` walkthrough
+
+We will walk through the [`basic_contract_ref`](TODO) example in order to demonstrate how
```
Comment (HCastano) [inline docs/basics/cross-contract-calling.md:33]: We need to wait for my other PR to get merged before filling this in

File: docs/basics/cross-contract-calling.md [hunk 2]
```diff
@@ -6,99 +6,345 @@ hide_title: true
...
-delegation and ink! messages can easily call other ink! messages.
-Given another ink! contract like, we can call any of its functions.
+In ink! contracts it is possible to call ink! messages and constructors of other
+on-chain contracts.
 
-See our [`delegator example contract`](https://github.com/paritytech/ink-examples/blob/main/delegator/lib.rs) 
-for an elaborate example which uses cross-contract calling.
+There are a few approaches to performing these cross-contract calls in ink!:
+1. Contract references (i.e `ContractRef`)
+1. Builders (i.e `CreateBuilder` and `CallBuilder`)
```
Comment (cmichi) [inline docs/basics/cross-contract-calling.md:16]:
```suggestion
1. Builders (i.e `CreateBuilder` and `CallBuilder`)

Contract references can only be used to do a cross-contract call to another ink!
contract, whereas the Builders approach can be used to issue a cross-contract
call to an arbitrary contract, independent of which languages it was written in ‒ be
it an ink! contract, a Solang one, or e.g. a contract written in ask!.
```

File: docs/basics/cross-contract-calling.md [hunk 3]
```diff
@@ -6,99 +6,345 @@ hide_title: true
...
 
-### How it Works
+## Contract References
 
-In order to deploy the delegator smart contract we first
-have to manually put the code of the other contract, receive
-its code hash from the signalled event and put their code hash
-into our calling smart contract.
+Contract references refer to structs which are generated by the ink! codegen for the
```
Comment (cmichi) [inline docs/basics/cross-contract-calling.md:20]:
```suggestion
Contract references refer to structs that are generated by the ink! code generation for the
```

File: docs/basics/cross-contract-calling.md [hunk 4]
```diff
@@ -6,99 +6,345 @@ hide_title: true
...
+[`ink::env::Error`](https://docs.rs/ink_env/latest/ink_env/enum.Error.html)
+and
+[`ink::LangError`](https://docs.rs/ink/latest/ink/enum.LangError.html)
+for more details on proper error handling.
+
+::: tip
+
+Because the `CallBuilder` requires only a contract's `AccountId` and message `selector`,
+we can call Solidity contracts compiled using the [Solang](TODO) compiler and deployed to
+a chain which supports the `contracts-pallet.`
```
Comment (cmichi) [inline docs/basics/cross-contract-calling.md:345]:
```suggestion
we can call Solidity contracts compiled using the [Solang](https://github.com/hyperledger/solang)
compiler and deployed to a chain that supports the `pallet-contracts`.
See [here](https://github.com/xermicus/call_solidity) for an example of how to do that.
```

**Review by cmichi — APPROVED** [review]
Once you've updated the PR, please also apply the changes to https://github.com/paritytech/ink-docs/blob/master/i18n/es/docusaurus-plugin-content-docs/current/basics/cross-contract-calling.md.

You should surround the English text with a `<div class="translateTodo">…</div>` tag and we'll then notify someone to translate them.

**Pull request merged**